### ISSUE TITLE
-DSEQAN_DISABLE_VERSION_CHECK missing from Seqan3

### ISSUE THREAD

**Issue by mr-c** [seqan/seqan3#1226]
For Debian we'd like a cmake-level directive to turn off version checking, thanks!

**Comment by h-2** [comment]
We have discussed this fairly long and we decided not to offer it, because the current setup is already very complex and hard to test (obdviously it is important for us not get this wrong as users will be unhappy if a request is made against their choice).

The current system:
  * the developer can disable for their app
  * the administrator or user can deactivate globally on their system via an environment variable
  * if none of the above happened, the user can disable via command-line (also before first run)
  * if none of that happened, the user still has to explicitly opt-in on their first interactive session

Only if none of the above applies **and** the user never once runs interactively, only non-interactively, then it will be performed in the background; but it will be logged very verbosely to cerr with information on how to deactivate it on the next run. I think this is a lot more privacy-friendly than what we had before and I would really rather not add more codepaths to this thing.

But if this makes Debian very unhappy we can discuss it again :wink:
